templates/libraries/okhttp-gson/pojo.mustache

```diff
@@ -553,6 +553,30 @@ public class {{classname}} {{#parent}}extends {{{.}}} {{/parent}}{{#vendorExtens
       }
       {{/required}}
       {{/isModel}}
+      {{#isEnum}}
+      {{#required}}
+      // validate the required field `{{{baseName}}}`
+      {{{datatypeWithEnum}}}.validateJsonElement(jsonObj.get("{{{baseName}}}"));
+      {{/required}}
+      {{^required}}
+      // validate the optional field `{{{baseName}}}`
+      if (jsonObj.get("{{{baseName}}}") != null && !jsonObj.get("{{{baseName}}}").isJsonNull()) {
+        {{{datatypeWithEnum}}}.validateJsonElement(jsonObj.get("{{{baseName}}}"));
```

Comment:
This is, I think, the only difference between the `isModel` block and the new enum blocks; through inspection of the template object I found that inline enums store the enum class name in `datatypeWithEnum`, not in `dataType`.